src/v/utils/named_type.h

```diff
@@ -137,6 +137,11 @@ class base_named_type<T, Tag, std::false_type> {
     explicit base_named_type(type&& v)
       : _value(std::move(v)) {}
 
+    template<typename... Args>
+    static base_named_type from(Args&&... args) {
+        return base_named_type(T(std::forward<Args>(args)...));
+    }
```

Comment:
How about something like (and I haven't tested this):
```c++
    template<typename... Args>
    requires std::constructible_from<T, Args...>
    explicit base_named_type(Args&&... args)
      : _value(std::forward<Args>(args)...) {}
```

Reply:
same question. i think generalizing the constructor would work out of the box?

Reply:
Great idea -- done